tests/unit/utils-request.js

```diff
@@ -0,0 +1,102 @@
+var simple = require('simple-mock')
+var test = require('tape')
+
+var request = require('../../lib/utils/request')
+
+test('request resolves when no error is given', function (t) {
+  t.plan(2)
+
+  simple.mock(request.internals, 'nets').callFn(function (options, cb) {
+    cb(null, {})
+  })
+
+  request({})
+
+    .catch(function () {
+      t.fail('should not be rejected')
+    })
+
+    .then(function () {
+      t.is(request.internals.nets.callCount, 1, 'nets request sent')
+      t.pass('promise should resolve')
+    })
+
+    .then(function () {
+      simple.restore()
+    })
+})
+
+test('request rejects with error', function (t) {
+  t.plan(4)
+
+  simple.mock(request.internals, 'nets').callFn(function (options, cb) {
+    cb({}, {})
```

Comment:
could you pass in an error as the first argument? I think the 2nd argument shouldn’t be necessary then:

```js
cb(new Error('oops'))
```